link: add Detach()

florianl opened the pull request:
Add [bpf_link_detach()](https://docs.ebpf.io/ebpf-library/libbpf/userspace/bpf_link__detach/) functionality.

mmat11 commented:
I see that `iter` links (and probably other) don't support `Detach()`; should we add a test for one of these cases? also, should we explicitly return `ErrNotSupported`?

florianl commented:
sorry for the delayed response :pray: 

> I see that iter links (and probably other) don't support Detach(); should we add a test for one of these cases?

With https://github.com/cilium/ebpf/pull/1830/commits/2520bb8e76b768d1281719270dc921eae98a083f I have added such a test.

> [..] should we explicitly return ErrNotSupported?

With https://github.com/cilium/ebpf/pull/1830/commits/c92c648d79aa04a3d47a8cd64a25cb21b73a0e4c I have added an error wrapper. Hope this is fine for you.

mmat11 commented:
lgtm! there are some failing CI jobs, not sure they are related

mmat11 reviewed (approved):
Sorry for the delay, I thought I approved already 🙈
